src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -124,6 +128,23 @@ public static float getBreakSpeed(EntityPlayer player, IBlockState state, float
         PlayerEvent.BreakSpeed event = new PlayerEvent.BreakSpeed(player, state, original, pos);
         return (MinecraftForge.EVENT_BUS.post(event) ? -1 : event.getNewSpeed());
     }
+    
+    public static boolean canSoilSustainPlant(IBlockAccess world, BlockPos pos, EnumFacing direction, IPlantable plantable, boolean canSustain)
+    {
+        if(world instanceof World){
+            CanSoilSustainPlantEvent event = new CanSoilSustainPlantEvent((World) world, pos, world.getBlockState(pos), plantable, direction, canSustain);
+            MinecraftForge.EVENT_BUS.post(event);
+            return event.canSustain();
+        } else{
+        	return canSustain;
+        }
+    }
+
+    public static float getGrowthChance(Block plant, World worldIn, BlockPos pos, float growthChance){
+        GetGrowthChanceEvent event = new GetGrowthChanceEvent(worldIn, pos, worldIn.getBlockState(pos.down()), (IPlantable) plant, growthChance);
```

Comment:
The position IS soil's position.
I can't pass IBlockState directly, because i didn't notice it in Block#canSustainPlant.

Anything else i should fix and push?